Silicon/NXP/iMX6Pkg/Drivers/PlatformSmbiosDxe/PlatformSmbiosDxe.c

```diff
@@ -617,50 +694,78 @@ SysInfoUpdateSmbiosType1 (
   )
 {
   CHAR16                *Family;
+  CHAR16                *FamilyOverride;
   GLOBAL_PAGE_DATA      *GlobalDataAddress;
   CHAR16                *Manufacturer;
+  CHAR16                *ManufacturerOverride;
   CHAR8                 *OptionalStrStart;
   CHAR16                *ProductName;
+  CHAR16                *ProductNameOverride;
   CHAR16                *SerialNumber;
   CHAR16                *SkuNumber;
+  CHAR16                *SkuNumberOverride;
   SMBIOS_TABLE_TYPE1    *SmbiosRecord;
   EFI_GUID              *SystemUuidFromPcd;
   EFI_GUID              *SystemUuid;
   CHAR16                *Version;
   UINT64                ProcessorSerialNumber;
   UINTN                 FamilyLen;
+  UINTN                 FamilyOverrideLen;
   UINT32                i;
   UINTN                 ManufacturerLen;
+  UINTN                 ManufacturerOverrideLen;
   UINTN                 ProductNameLen;
+  UINTN                 ProductNameOverrideLen;
   UINTN                 SerialNumberLen;
   UINTN                 SkuNumberLen;
+  UINTN                 SkuNumberOverrideLen;
   EFI_SMBIOS_HANDLE     SmbiosHandle;
   UINT32                SmbiosRecordLen;
   EFI_STATUS            Status;
   UINTN                 VersionLen;
 
+  FamilyOverride = NULL;
+  ManufacturerOverride = NULL;
+  ProductNameOverride = NULL;
+  SkuNumberOverride = NULL;
   SerialNumber = NULL;
   SmbiosRecord = NULL;
   SystemUuid = NULL;
   SmbiosRecordLen = sizeof (SMBIOS_TABLE_TYPE1);
 
   // 04h - Manufacturer String
-  Manufacturer = (CHAR16 *)FixedPcdGetPtr (PcdSystemManufacturer);
-  ManufacturerLen = StrLen (Manufacturer);
-  if (ManufacturerLen == 0) {
-    DEBUG ((DEBUG_ERROR, "%a: PcdSystemManufacturer not filled\n", __FUNCTION__));
-    Status = EFI_INVALID_PARAMETER;
-    goto Exit;
+  Status = RetrieveSmbiosVariable (L"System Manufacturer",

```

Comment:
Should the "System Manufacturer" and other const string identifiers themselves be defined as a pcd value, or defined in a header file?

Reply:
Yes. Common header makes sense here. Will fix